ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java

```diff
@@ -135,26 +133,11 @@ public void setBackground(Drawable drawable) {
         "This method is not supported for ReactViewGroup instances");
   }
 
-  public void setNativeBackground(@Nullable ReadableMap nativeBackground) {
-    mNativeBackground = nativeBackground;
-    refreshTranslucentBackgroundDrawable();
-  }
-
-  public void refreshTranslucentBackgroundDrawable() {
+  public void setTranslucentBackgroundDrawable(@Nullable Drawable background) {
     // it's required to call setBackground to null, as in some of the cases we may set new
     // background to be a layer drawable that contains a drawable that has been previously setup
     // as a background previously. This will not work correctly as the drawable callback logic is
     // messed up in AOSP
-
-    Drawable background = null;
```

Comment:
This is where the logic flaw lives -- this changes how background is calculated. With this change it is *only* calculated from a background set via setNativeBackground. However, there are other ways that the background could be set.

Specifically, in ReactViewGroup#getOrCreateReactViewBackground() the background property is being set through a means other than setNativeBackground.